src/algokit/cli/tasks/analyze.py

```diff
@@ -0,0 +1,230 @@
+        "the last report. Reports are stored in the "
+        ".algokit/static-analysis/snapshots folder by default. Use --output for a "
+        "custom path."
+    ),
+)
+@click.option(
+    "-o",
+    "--output",
+    "output_path",
+    required=False,
+    default=None,
+    type=click.Path(dir_okay=True, file_okay=False, resolve_path=True, path_type=Path),
+    help=(
+        "Directory path where to store the results of the static analysis. "
+        "Defaults to .algokit/static-analysis/snapshots."
+    ),
+)
+@click.option(
+    "-e",
+    "--exclude",
+    "detectors_to_exclude",
+    multiple=True,
+    default=[],
+    type=click.STRING,
+    help="Exclude specific vulnerabilities from the analysis. Supports multiple exclusions in a single run.",
+)
+def analyze(  # noqa: PLR0913
+    *,
+    input_paths: tuple[Path],
+    recursive: bool,
+    force: bool,
+    diff_only: bool,
+    output_path: Path | None,
+    detectors_to_exclude: list[str],
+) -> None:
+    """
+    Analyze TEAL programs for common vulnerabilities using Tealer.
+    """
+
+    detectors_to_exclude = sorted(set(detectors_to_exclude))
+    input_files = get_input_files(input_paths=input_paths, recursive=recursive)
+
+    if not force:
+        click.confirm(
+            click.style(
+                "Warning: This task uses `tealer` to suggest improvements for your TEAL programs, "
+                "but remember to always test your smart contracts code and follow modern "
+                "software engineering practices. Do you understand?",
+                fg="yellow",
+            ),
+            default=True,
+            abort=True,
+        )
+
+    reports = {}
+    duplicate_files: dict[str, int] = {}
+    prepare_artifacts_folders(output_path)
+    total_files = len(input_files)
+    for index in range(total_files):
+        cur_file = input_files[index]
+        file = cur_file.resolve()
+
+        if has_template_vars(file):
+            click.secho(
+                f"Warning: Skipping {file} due to template variables. Substitute them before scanning.",
+                err=True,
+                fg="yellow",
+            )
+            continue
+
+        filename = generate_report_filename(file, duplicate_files)
+
+        # If diff_only is enabled, store the report in the snapshots folder otherwise store it in the reports folder
+        # If a custom output path is provided, store the report in the specified path
+        report_output_root = output_path or TEALER_SNAPSHOTS_ROOT
+        report_output_path = report_output_root / filename
+
+        command = generate_tealer_command(cur_file, report_output_path, detectors_to_exclude)
+        old_report = load_tealer_report(str(report_output_path)) if report_output_path.exists() and diff_only else None
+        try:
+            run_with_animation(run_tealer, f"Analyzing {index + 1} out of {total_files} files", command)
+
+            if diff_only and old_report:
```

Comment:
Currently if we have `diff_only = True` and `old_report=None`, the `has_baseline_diff` will not be run and as a result the process will exit with code 0. Essentially it will behave as if the `--diff` flag was not supplied.

I think we want to exit with code 1 if `diff_only = True` and `old_report=None` and not even run tealer in this instance.